### ISSUE TITLE
UnhandledPromiseRejectionWarning: TypeError: abi.map is not a function

### ISSUE THREAD

**Issue by Gajesh2007** [solana-labs/solana-solidity.js#11]
I was trying to deploy the smart contract on solana but while deploying im getting the below error:

```
Connecting to your local Solana node ...
Airdropping SOL to a new wallet ...
PublicKey {
  _bn: <BN: 6a1dddb4942d3ed20116fb801ee73d6ef5e9ec3c805435f96f9cec2a1be5b1c1>
}
(node:30305) UnhandledPromiseRejectionWarning: TypeError: abi.map is not a function
    at new Interface (/node_modules/@ethersproject/abi/lib/interface.js:101:60)
    at new Contract (node_modules/@solana/solidity/dist/esm/src/contract.js:21:20)
    at /ERC20/project/erc20.js:26:22
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:30305) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:30305) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

ERC20.js:
```
const { Connection, LAMPORTS_PER_SOL, Keypair } = require('@solana/web3.js');
const { Contract, publicKeyToHex } = require('@solana/solidity');
const { readFileSync } = require('fs');

const ERC20_ABI = JSON.parse(readFileSync('./build/ERC20.abi', 'utf8'));
const BUNDLE_SO = readFileSync('./build/bundle.so');

(async function () {
    console.log('Connecting to your local Solana node ...');
    const connection = new Connection('https://api.devnet.solana.com', 'confirmed');

    const payer = Keypair.generate();
    while (true) {
        console.log('Airdropping SOL to a new wallet ...');
        await connection.requestAirdrop(payer.publicKey, 1 * LAMPORTS_PER_SOL);
        await new Promise((resolve) => setTimeout(resolve, 1000));
        if (await connection.getBalance(payer.publicKey)) break;
    }

    console.log(payer.publicKey);

    const address = '0x' + payer.publicKey.toBuffer().toString('hex');
    const program = Keypair.generate();
    const storage = Keypair.generate();

    const contract = new Contract(
        connection,
        program.publicKey,
        storage.publicKey,
        ERC20_ABI,
        payer
    );

    console.log('Deploying the Solang-compiled ERC20 program ...');
    await contract.load(program, BUNDLE_SO);

    console.log('Program deployment finished, deploying the ERC20 contract ...');
    await contract.deploy(
        'ERC20',
        ['Solana', 'SOL', '1000000000000000000'],
        program,
        storage,
        4096 * 8
    );

    console.log('Contract deployment finished, invoking some contract functions ...');
    const symbol = await contract.symbol();
    const balance = await contract.balanceOf(address);

    console.log(`ERC20 contract for ${symbol} deployed!`);
    console.log(`Your wallet at ${address} has a balance of ${balance} tokens.`);

    contract.addEventListener(function (event) {
        console.log(`${event.name} event emitted!`);
        console.log(`${event.args[0]} sent ${event.args[2]} tokens to ${event.args[1]}`);
    });

    console.log('Sending tokens will emit a "Transfer" event ...');
    const recipient = Keypair.generate();
    await contract.transfer(publicKeyToHex(recipient.publicKey), '1000000000000000000');

    process.exit(0);
})();


```

**Comment by seanyoung** [comment]
We are just re-working some parts of the library. I'll get back to you soon.

**Comment by Gajesh2007** [comment]
👍

**Comment by Gajesh2007** [comment]
following up :D

**Comment by seanyoung** [comment]
Sorry, I was at the solana conference in Lisbon. I've just come back. Time to catch up!

**Comment by Gajesh2007** [comment]
![image](https://thumbs.gfycat.com/OblongUntriedCow-small.gif)

**Comment by jordansexton** [comment]
@seanyoung I think this and #12 are both due to the repo and the published npm package being out of sync.

**Comment by seanyoung** [comment]
Yes, we really need to publish a new version. I've almost got all the test running.

**Comment by seanyoung** [comment]
So I've just published 0.0.17 - should be fixed in that! Sorry about the delay

**Comment by jordansexton** [comment]
Closing, it sounds like this is fixed in the current release.
